### PR TITLE
more metadata fix

### LINKED ISSUE #47: need subject metadata for old experiments
specifically:
- sex
- species
- genotype
- description (where the animal came from)
- weight

### LINKED ISSUE #53: add unit to subject weights in metadata
> Is the weight unit pre-defined in nwb? The units of weight are in grams

_Originally posted by @jthermiz in https://github.com/BouchardLab/nsds_lab_to_nwb/issues/46#issuecomment-859986965_

### LINKED ISSUE #55: do not include ecog/poly device metadata when the device was not used
For this block, poly was false so should we not include the poly device details?
In the future if one uses poly but not ecog should we not include ecog device details?

_Originally posted by @jthermiz in https://github.com/BouchardLab/nsds_lab_to_nwb/issues/46#issuecomment-859986965_

### LINKED ISSUE #66: update device filtering description in metadata
Specifically, update the text here:
https://github.com/BouchardLab/nsds_lab_to_nwb/blob/b7c5c269bcdae8680f10161cd9cab509a5ee8e5b/nsds_lab_to_nwb/metadata/metadata_manager.py#L97-L99

Filtering details were discussed in issue #49.

--------------

For reference, this information will be used in the ElectrodeOriginator as in
https://github.com/BouchardLab/nsds_lab_to_nwb/blob/b7c5c269bcdae8680f10161cd9cab509a5ee8e5b/nsds_lab_to_nwb/components/electrode/electrodes_originator.py#L34

### PR DESCRIPTION
**Pull request by jihyunbak**
# Description and related issues

This PR fixes various metadata issues.

Closes #55
Closes #66 
Closes #53
Resolves already closed issue #48 
Closes #47

# Checklist:

- [ ] All tests pass on catscan: run `pytest --basetemp=tmp -sv -n 8 tests` on catscan from the root directory
- [ ] If needed, docs have been update: `docs/source` has been updated for any added, moved, or removed files
- [x] Docs build with no errors: run `make clean & make html` from the `docs` folder
- [ ] No python formatting errors: run `flake8 nsds_lab_to_nwb tests` from the root directory
    - [x] cleared for files changed in this PR only

### CODE REVIEW

**Comment by jihyunbak** [comment]
This PR is ready for review again. @JesseLivezey @jthermiz

File: nsds_lab_to_nwb/metadata/metadata_manager.py [hunk 1]
```diff
@@ -69,6 +71,14 @@ def common_check(self):
         '''
         if 'subject' not in self.metadata_input:
             self.metadata_input['subject'] = {}
+            self.metadata_input['subject']['subject_id'] = self.animal_name
+
+        # fix subject weight unit - always 'g' in our case
+        subject_metadata = self.metadata_input['subject']
+        if 'weight' in subject_metadata:
+            weight = str(subject_metadata['weight'])
+            if weight[-1] != 'g':
```
Comment (JesseLivezey) [inline nsds_lab_to_nwb/metadata/metadata_manager.py:80]: I'm not clear on how this field is created, but maybe it's better to check `if 'g' not in weight`:? For instance, that would catch if a trailing space was entered after the 'g' or something like that.

**Review by JesseLivezey — COMMENTED** [review]
One suggestion, but otherwise, LGTM.

Could you run
```
pytest --basetemp=tmp -sv -n 8 tests/test_catscan.py
```
and paste which tests fail?

File: nsds_lab_to_nwb/metadata/metadata_manager.py (hunk 1 above)
Comment (jihyunbak) [reply nsds_lab_to_nwb/metadata/metadata_manager.py]: Good point, pushed a fix commit!

**Comment by jihyunbak** [comment]
@JesseLivezey  Here's the output from `test_catscan.py`. Lines are hard truncated to my window width...

```
==================================== short test summary info =====================================
FAILED test_catscan.py::test_nwb_builder[RVG16_B04] - FileNotFoundError: [Errno 2] No such file...
FAILED test_catscan.py::test_nwb_builder[RVG16_B02] - FileNotFoundError: [Errno 2] No such file...
FAILED test_catscan.py::test_nwb_builder[RVG16_B05] - FileNotFoundError: [Errno 2] No such file...
FAILED test_catscan.py::test_nwb_builder[RVG16_B09] - ValueError: Stimulus type 'nan' not found.
FAILED test_catscan.py::test_nwb_builder[RVG16_B10] - ValueError: Unknown stimulus type '{stim_...
FAILED test_catscan.py::test_nwb_builder[RVG16_B07] - OSError: Unable to open file (unable to o...
FAILED test_catscan.py::test_nwb_builder[RVG16_B06] - OSError: Unable to open file (unable to o...
FAILED test_catscan.py::test_nwb_builder[RVG16_B08] - FileNotFoundError: [Errno 2] No such file...
FAILED test_catscan.py::test_legacy_nwb_builder[R56_B10] - OSError: Unable to open file (unable...
============================ 9 failed, 3 passed in 140.03s (0:02:20) ============================
```

**Pull request merged**